session_rpcserver.go

```diff
@@ -143,18 +143,48 @@ func (s *sessionRpcServer) AddSession(_ context.Context,
 		}
 
 		for _, op := range req.MacaroonCustomPermissions {
-			if op.Entity == macaroons.PermissionEntityCustomURI {
-				_, ok := s.cfg.permMgr.URIPermissions(op.Action)
+			if op.Entity != macaroons.PermissionEntityCustomURI {
+				permissions = append(permissions, bakery.Op{
+					Entity: op.Entity,
+					Action: op.Action,
+				})
+
+				continue
+			}
+
+			// First check if this is a wildcard URI.
+			uris, isWildCard := s.cfg.permMgr.ExpandWildCardURI(
+				op.Action,
+			)
+			if !isWildCard {
```

Comment:
nit: invert this `if` so the wildcard logic comes directly after the `ExpandWildCardURI` and the default case is non-indented?